### PR TITLE
[vectorization](storage)support seq column in storage layer

### LINKED ISSUE #8185: [Feature] Support seq column in segment comparison
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Support seq column in segment comparison

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangbo**
# Proposed changes

Issue Number: close #8185

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/olap/vgeneric_iterators.cpp [hunk 1]
```diff
@@ -166,6 +166,15 @@ class VMergeIteratorContext {
         if (cmp_res != 0) {
             return cmp_res > 0;
         }
+        
+        if (_sequence_id_idx != -1) {
+            int col_cmp_res = this->_block.compare_column_at(_index_in_block, rhs._index_in_block, _sequence_id_idx, rhs._block, -1);
+            auto left_col = this->_block.get_by_position(_sequence_id_idx).column;
```
Comment (HappenLee) [inline be/src/vec/olap/vgeneric_iterators.cpp:172]: the `left_col` and `right_col` look like unless， maybe not need to add method of `_block.compare_column_at`. direct use left_col method `compare_at` function look more pretty？

File: be/src/vec/core/block.h [hunk 2]
```diff
@@ -273,6 +273,14 @@ class Block {
         return 0;
     }
 
+    //note(wb) no DCHECK here, because this method is only used after compare_at now, so no need to repeat check here.
+    // If this method is used in more places, you can add DCHECK case by case.
+    int compare_column_at(size_t n, size_t m, size_t col_idx, const Block& rhs, int nan_direction_hint) const {
```
Comment (HappenLee) [inline be/src/vec/core/block.h:278]: why not directly use the coiumn method  `compare_at` ?

File: be/src/vec/olap/vgeneric_iterators.cpp (hunk 1 above)
Comment (wangbo) [reply be/src/vec/olap/vgeneric_iterators.cpp:172]: 👌🏻

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**